### PR TITLE
[FIX] Parcelize annotation on State classes

### PR DESCRIPTION
**Pull request by francofusaro**
## Motivación y Contexto
We are fixing [PXN-2829](https://mercadolibre.atlassian.net/browse/PXN-2829) where it was crashing in some cases when checkout activity was destroyed while on lock screen activity. We also fixed other state that was having the same problem and we expect to be fixing [other NPE](https://app.bugsnag.com/mercadolibre/mercado-pago-android/errors/610f742998a57c0007e48b3c?filters[event.since]=7d&filters[error.status]=open&filters[search]=android.px) with this.

## Descripción
Parcelize annotation requires all fields that will be stored/restored to be in a primary constructor so we moved those that were not on PayButtonViewModel state and ExpressPaymentFragment state.

## Cómo probarlo
The way to make it crash was to mock a fingerprint activity to testapp (replacing current default security behavior on px configurer), enable destroy activities on emulator, start a checkout, click pay, when we are on our fingerprint activity then send the app to background and come back, that was not restoring our paymentConfiguration so it was null.

## Compartir conocimiento
https://developer.android.com/kotlin/parcelize 

> @Parcelize requires all serialized properties to be declared in the primary constructor. The plugin issues a warning on each property with a backing field declared in the class body. Also, you can't apply @Parcelize if some of the primary constructor parameters are not properties.

### CODE REVIEW

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/ExpressPaymentState.kt [hunk 1]
```diff
@@ -5,7 +5,7 @@ import com.mercadopago.android.px.internal.viewmodel.SplitSelectionState
 import kotlinx.android.parcel.Parcelize
 
 @Parcelize
-internal class ExpressPaymentState: BaseState {
-    var paymentMethodIndex = 0
-    var splitSelectionState = SplitSelectionState()
-}
+internal class ExpressPaymentState @JvmOverloads constructor(
+    var paymentMethodIndex: Int = 0,
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/ExpressPaymentState.kt:9]: Parecen inferibles estos tipos

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/payment_result/remedies/RemediesViewModel.kt [hunk 2]
```diff
@@ -217,7 +217,5 @@ internal class RemediesViewModel(
     }
 
     @Parcelize
-    class State(var paymentRecovery: PaymentRecovery) : BaseState {
-        var cvv = ""
-    }
+    class State(var paymentRecovery: PaymentRecovery, var cvv: String = "") : BaseState
```
Comment (guchito9) [inline px-checkout/src/main/java/com/mercadopago/android/px/internal/features/payment_result/remedies/RemediesViewModel.kt:220]: Este tipo también es inferible no? Pero veo que los pusiste a propósito :P

Otra cosa, no sé si es "): BaseState" o ") : BaseState" en cuánto al espacio. Habría que tirarle un autoformat

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/ExpressPaymentState.kt (hunk 1 above)
Comment (francofusaro) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/ExpressPaymentState.kt:9]: can't infer those types :(
[here](https://discuss.kotlinlang.org/t/why-can-not-type-of-function-parameters-with-default-value-be-omited/3641/2) it's explained and they say it's something about compiling performance :/

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/payment_result/remedies/RemediesViewModel.kt (hunk 2 above)
Comment (francofusaro) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/payment_result/remedies/RemediesViewModel.kt:220]: Oops, fixing the whitespace, forgot to autoformat 🤣

**Review by jorGonzalez291292 — DISMISSED** [review]

**Review by guchito9 — APPROVED** [review]

**Review by jvillada31 — APPROVED** [review]
LG, just one comment given that the State classes now have a primary constructor I think that we could make them `data classes`

**Review by guchito9 — APPROVED** [review]

**Review by jorGonzalez291292 — APPROVED** [review]

**Review by jvillada31 — APPROVED** [review]

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/pay_button/PayButtonViewModel.kt [hunk 3]
```diff
@@ -323,10 +323,10 @@ internal class PayButtonViewModel(
     }
 
     @Parcelize
-    class State : BaseState {
-        var paymentConfiguration: PaymentConfiguration? = null
-        var paymentModel: PaymentModel? = null
-        var retryCounter = 0
-        var observingService = false
-    }
+    data class State(
```
Comment (murijr) [inline px-checkout/src/main/java/com/mercadopago/android/px/internal/features/pay_button/PayButtonViewModel.kt:326]: No crees que sería bueno tener un estado inmutable ? De esa forma evitaremos la inconsistencia de la UI y los problemas de subprocesos, ¿no crees? A medida que la clase State se convierte en una clase de datos, podemos usar copy para emitir un nuevo estado. Que crees ?

Comment (francofusaro) [reply px-checkout/src/main/java/com/mercadopago/android/px/internal/features/pay_button/PayButtonViewModel.kt:326]: Maybe, but for doing that we should make a wider analysis of where and how those states are used, here we are focusing in fixing this bug that is causing lots of problems. Anyways, for sure we should take that in count for future improvements, it's a good point 👌

**Pull request merged**